stumpy/floss.py

```diff
@@ -514,6 +551,17 @@ def __init__(
                 - core.z_norm(right_nn, 1),
                 axis=1,
             )
+            nn_subseq_isconstant = self._T_subseq_isconstant[
+                self._mp[:, 3].astype(np.int64)
+            ]
```

Comment:
As discussed in #808, we may add a comment to inform users of the fact that some of the arrays in `right_nn` and some booean values in `nn_subseq_isconstant` are meaningless as `self._mp[:, 3]` contains `-1`, which is not the start index of  a subsequence.  Howver, the postprocessing at the end of this "if/else block" takes care of this matter.

@seanlaw : Any suggestion?

Reply:
Is it to inform users or to inform contributors? I think we should be consistent and follow what we said earlier by adding a comment/reminder

Reply:
@seanlaw 
Oops! I meant `contributors`. I am going to use what you said in #808 

```
"Technically, a negative index value has no meaning (i.e., it is an invalid index). "
"However, in numpy, a negative value corresponds to the last position of an array "
"and so we can't "trust" that negative index value anymore and we need to replace it "
"with something else that is not -1."
```

I may do some minor changes if need. But, the main message will follow yours.